x-pack/plugin/build.gradle

```diff
@@ -104,39 +104,26 @@ integTestRunner {
   systemProperty 'tests.rest.blacklist', blacklist.join(',')
 }
 
-// location of generated keystores and certificates
+// location for keys and certificates
 File keystoreDir = new File(project.buildDir, 'keystore')
-
-// Generate the node's keystore
-File nodeKeystore = new File(keystoreDir, 'test-node.jks')
-task createNodeKeyStore(type: LoggedExec) {
-  doFirst {
-    if (nodeKeystore.parentFile.exists() == false) {
-      nodeKeystore.parentFile.mkdirs()
-    }
-    if (nodeKeystore.exists()) {
-      delete nodeKeystore
+File nodeKey = new File(keystoreDir, 'testnode.pem')
+File nodeCert = new File(keystoreDir, 'testnode.crt')
+
+// Add key and certs to test classpath: it expects them there
+// User cert and key PEM files instead of a JKS Keystore for the cluster's trust material so that
+// it can run in a FIPS 140 JVM
+task copyKeyCerts(type: Copy) {
+  from('./core/src/test/resources/org/elasticsearch/xpack/security/transport/ssl/certs/simple/') {
```

Comment:
Creating coupling between projects like this is not a good idea in Gradle.
We should at least reference this as somethign like `project(':x-pack:plugin:core').file('src/test/resources/org/elasticsearch/xpack/security/transport/ssl/certs/simple/')` to make this more explicit. 
The clean solution would be to have a small Gradle project with these resources and then 
```
   dependencies {
         testRuntime project(':test:fips:certs') 
    }
```
Where `:test:fips:certs`  would just have these files that need to be on the CP. 
Gradle will take care  of dependencies  and adding it to the cp automatically and it will play nicer with `--parallel` as well. Would also mean that the copy task doesn't have to be repeated.

In this case, would it be feasible to have the build generate these files ?
Maybe even with random passwords and such ? Rest integration tests don't run in the IDE anyhow.
Might prevent some automated checks for committed private keys from tripping on our repo - even if it's only for testing.

Reply:
> In this case, would it be feasible to have the build generate these files ?

Unfortunately not. `keytool` only generates Private Keys and stores them in a Keystore, it can't output them as a PEM file. So we can only generate a PKCS12 or a JKS keystore containing the private key and associated certificate. `keytool` also doesn't support exporting private keys from a keystore. The only solution would be to use an external tool like `openssl` to either export the key and certificate form a PKCS12 store or generate the key , certificate in the first place. We can't be sure that `openssl` will be available on all platforms though ( and I don't think Windows have a utility with corresponding functionality ) 

> The clean solution would be to have a small Gradle project with these resources and then

Not all projects need all files ( `test{node,client}.{jks,pem,crt}` ) but it wouldn't hurt to add them. No issue going with this suggestion

Reply:
How about doing in Java what openssl would do ? Maybe using BouncyCastle or some other lib ? Would it be more effort than what it's worth ?

Reply:
Nope, can't do that. We explicitly removed BC dependency recently on purpose ( see https://github.com/elastic/elasticsearch/pull/30358  on why and why). 
I guess I can look into the possibility of exporting the keys in Java (without external deps)  but in my mind this is much more complicated and I can't really see the benefits. 
Granted that the files are referenced correctly or even replicated in this Gradle project so that we don't need to reference them from `x-pack:plugin:core`, or even better I go down the path of the new gradle project with those files as resources, do you have any specific concerns?

Reply:
In this case, BouncyCastle would be a dependency for the build, which is a bit different, I don't think we'll be able to run Gradle itself on a FIPS JVM, of if we even want to.  That being said I suggested it because it  seemed the cleanest and not too much work - but admittedly I'm missing a lot of context on this so I could be totally wrong. I don't have a strong preference for this, separate project or referencing trough project is also fine for me ( in this order of preference ).